src/cache/requests/scalar/item_get_type.rs

```diff
@@ -0,0 +1,152 @@
+            Some(item_get_type_response::Result::Found(found)) => Ok(ItemGetType::Hit {
+                key_type: match found.item_type() {
+                    momento_protos::cache_client::item_get_type_response::ItemType::Scalar => {
+                        ItemType::Scalar
+                    }
+                    momento_protos::cache_client::item_get_type_response::ItemType::Dictionary => {
+                        ItemType::Dictionary
+                    }
+                    momento_protos::cache_client::item_get_type_response::ItemType::List => {
+                        ItemType::List
+                    }
+                    momento_protos::cache_client::item_get_type_response::ItemType::Set => {
+                        ItemType::Set
+                    }
+                    momento_protos::cache_client::item_get_type_response::ItemType::SortedSet => {
+                        ItemType::SortedSet
+                    }
+                },
+            }),
+            _ => unreachable!(),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum ItemType {
+    Scalar = 0,
+    Dictionary = 1,
+    List = 2,
+    Set = 3,
+    SortedSet = 4,
+}
+
+/// Response for a get item type operation.
+///
+/// If you'd like to handle misses you can simply match and handle your response:
+/// ```
+/// # use momento::MomentoResult;
+/// # use momento::cache::{ItemGetType, ItemType};
+/// use std::convert::TryInto;
+/// # let response = ItemGetType::Hit { key_type: ItemType::Scalar };
+/// let item: ItemType = match response {
+///     ItemGetType::Hit { key_type } => key_type.try_into().expect("Expected an item type!"),
+///     ItemGetType::Miss => return // probably you'll do something else here
+/// };
+/// ```
+///
+/// You can cast your result directly into a Result<ItemType, MomentoError> suitable for
+/// ?-propagation if you know you are expecting a GetItemType::Hit.
+///
+/// Of course, a Miss in this case will be turned into an Error. If that's what you want, then
+/// this is what you're after:
+/// ```
+/// # use momento::MomentoResult;
+/// # use momento::cache::{ItemGetType, ItemType};
+/// use std::convert::TryInto;
+/// # let response = ItemGetType::Hit { key_type: ItemType::Scalar };
+/// let itemType: MomentoResult<ItemType> = response.try_into();
+/// ```
+#[derive(Debug, PartialEq, Eq)]
+pub enum ItemGetType {
+    Hit { key_type: ItemType },
+    Miss,
+}
+
+impl TryFrom<ItemGetType> for ItemType {
+    type Error = MomentoError;
+
+    fn try_from(value: ItemGetType) -> Result<Self, Self::Error> {
+        match value {
+            ItemGetType::Hit { key_type } => Ok(key_type),
+            ItemGetType::Miss => Err(MomentoError {
+                message: "item get type response was a miss".into(),
+                error_code: MomentoErrorCode::Miss,
+                inner_error: None,
+                details: None,
+            }),
```

Comment:
tried to make this match the `get` response `into`/`from` accessors as much as possible